pandas/tests/io/test_excel.py

```diff
@@ -605,30 +553,28 @@ def test_read_from_http_url(self, ext):
         url = ('https://raw.github.com/pandas-dev/pandas/master/'
                'pandas/tests/io/data/test1' + ext)
         url_table = read_excel(url)
-        local_table = self.get_exceldf('test1', ext)
+        local_table = read_excel('test1' + ext)
         tm.assert_frame_equal(url_table, local_table)
 
     @td.skip_if_not_us_locale
     def test_read_from_s3_url(self, ext, s3_resource):
         # Bucket "pandas-test" created in tests/io/conftest.py
-        file_name = os.path.join(self.dirpath, 'test1' + ext)
-
-        with open(file_name, "rb") as f:
+        with open('test1' + ext, "rb") as f:
             s3_resource.Bucket("pandas-test").put_object(Key="test1" + ext,
                                                          Body=f)
 
         url = ('s3://pandas-test/test1' + ext)
         url_table = read_excel(url)
-        local_table = self.get_exceldf('test1', ext)
+        local_table = read_excel('test1' + ext)
         tm.assert_frame_equal(url_table, local_table)
 
     @pytest.mark.slow
     # ignore warning from old xlrd
     @pytest.mark.filterwarnings("ignore:This metho:PendingDeprecationWarning")
-    def test_read_from_file_url(self, ext):
+    def test_read_from_file_url(self, ext, datapath):
 
         # FILE
-        localtable = os.path.join(self.dirpath, 'test1' + ext)
+        localtable = os.path.join(datapath("io", "data"), 'test1' + ext)
```

Comment:
This was actually failing with the fixturized approach because the URL requires an absolute path, or else it raises a URLError and gets skipped. 

Quite a few ways to do this but I figured just reusing the datapath fixture was easiest, especially since this is pending deprecation